multiqc/modules/nanoq/nanoq.py

```diff
@@ -0,0 +1,303 @@
+"""MultiQC module to parse output from nanoq"""
+
+import re
+import logging
+from typing import Dict, List, Any
+
+from multiqc import config
+from multiqc.base_module import BaseMultiqcModule, ModuleNoSamplesFound
+from multiqc.plots import bargraph
+from multiqc.utils import mqc_colour
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+
+class MultiqcModule(BaseMultiqcModule):
+    """Nanoq module"""
+
+    _KEYS_MAPPING = {
+        "Number of reads": "Number of reads",
+        "Number of bases": "Number of bases",
+        "N50 read length": "N50 read length",
+        "Longest read": "Longest read",
+        "Shortest read": "Shortest read",
+        "Mean read length": "Mean read length",
+        "Median read length": "Median read length",
+        "Mean read quality": "Mean read quality",
+        "Median read quality": "Median read quality",
+    }
+
+    _stat_types = ("summary", "quality")
+
+    def __init__(self):
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(
+            name="nanoq",
+            anchor="nanoq",
+            href="https://github.com/nerdna/nanoq/",
+            info="reports read quality and length from nanopore sequencing data in fast{a,q}.{gz,bz2,xz} format.",
+            doi="10.21105/joss.02991",
+        )
+
+        # Find and load any nanoq reports
+        self.nanoq_summary: Dict[str, Dict[str, float]] = {}
+        self.nanoq_data: Dict[str, Dict[str, float]] = {}
```

Comment:
We started to change this practice and avoiding adding any new class attributes to MultiqcModule. Would be nice to refactor by keeping `nanoq_data` etc local variables. I can do that in your PR for you to take a look